parser/src/main/java/com/facebook/coresql/lint/LintVisitor.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.coresql.lint;
+
+import com.facebook.coresql.parser.AndExpression;
+import com.facebook.coresql.parser.AstNode;
+import com.facebook.coresql.parser.OrExpression;
+import com.facebook.coresql.parser.SimpleNode;
+import com.facebook.coresql.parser.SqlParserDefaultVisitor;
+import com.facebook.coresql.warning.CoreSqlWarning;
+import com.facebook.coresql.warning.WarningCollector;
+
+import java.util.List;
+
+import static com.facebook.coresql.warning.StandardWarningCode.MIXING_AND_OR_WITHOUT_PARENTHESES;
+
+/**
+ * A visitor that validates an AST built from an SQL string. Right now, it validates
+ * a single clause: don't mix AND and OR without parentheses.
+ */
+public class LintVisitor
```

Comment:
Lint is usually a collection of rules. So this visitor should be named for the specific rule - something like MixedAndOR